content/docs/rfcs/26/README.md

```diff
@@ -0,0 +1,152 @@
+---
+slug: 26
+title: 26/WAKU-PAYLOAD
+name: Waku Message Payload Encryption
+status: draft
+editor: Oskar Thoren <oskar@status.im>
+contributors:
+---
+
+This specification describes how Waku provides provides confidentiality, authenticity, and integrity, as well as some form unlinkability.
+Specifically, it describes how encryption, decryption and signing works in [6/WAKU1](/spec/6) and in [10/WAKU2](/spec/10) with [14/WAKU-MESSAGE version 1](/spec/14/#version1).
+
+This specification effectively replaces [7/WAKU-DATA](/spec/7) as well as [6/WAKU1 Payload encryption](/spec/6/#payload-encryption) but written in a way that is agnostic and self-contained for Waku v1 and Waku v2.
+
+Large sections of the specification originate from [EIP-627: Whisper spec](https://eips.ethereum.org/EIPS/eip-627) as well from [RLPx Transport Protocol spec (ECIES encryption)](https://github.com/ethereum/devp2p/blob/master/rlpx.md#ecies-encryption) with some modifications.
+
+## Design requirements
+
+- *Confidentiality*: The adversary should not be able to learn what data is being sent from one Waku node to one or several other Waku nodes.
+- *Authenticity*: The adversary should not be able to cause Waku endpoint to accept data from any third party as though it came from the other endpoint.
+- *Integrity*: The adversary should not be able to cause a Waku endpoint to accept data that has been tampered with.
+
+Notable, *forward secrecy* is not provided for at this layer.
+If this property is desired, a more fully featured secure communication protocol can be used on top, such as [Status 5/SECURE-TRANSPORT](https://specs.status.im/spec/5).
+
+It also provides some form of *unlinkability* since:
+- only participants who are able to decrypt a message can see its signature
+- payload are padded to a fixed length
+
+## Cryptographic primitives
+
+- AES-256-GCM
+- ECIES
+- ECSDA
+- KECCAK-256
+
+ECIES is using the following cryptosystem:
+- Curve: secp256k1
+- KDF: NIST SP 800-56 Concatenation Key Derivation Function
+- MAC: HMAC with SHA-256
+- AES: AES-128-CTR
+
+## Specification
+
+For 6/WAKU1, the `data` field is used in the `waku envelope`, and the field MUST contain the encrypted payload.
+
+For 10/WAKU2, the `payload` field is used in `WakuMessage` and MUST contain the encrypted payload.
```

Comment:
Since this replaces https://rfc.vac.dev/spec/14/#payload-encryption, it may get lost that payload encryption remains optional for `10/WAKU2`.

Reply:
True. I actually think we could make it optional here too, that's the original behavior.

Reply:
```suggestion
For 10/WAKU2, the `payload` field is used in `WakuMessage` and SHOULD contain the encrypted payload.
```

Reply:
@jm-clius changing to a SHOULD, wdyt?

Reply:
LGTM

Reply:
MAY, fixed